providers/efa/verbs.c

```diff
@@ -1397,6 +1397,11 @@ static inline void efa_sq_ring_doorbell(struct efa_sq *sq, uint16_t pc)
 {
 	mmio_flush_writes();
 	mmio_write32(sq->wq.db, pc);
+	/* Make sure that no further writes to LLQ BAR are performed before
+	 * the DB write is done. Otherwise we might get more than max tx batch
+	 * writes to LLQ between two doorbells.
+	 */
+	mmio_flush_writes();
```

Comment:
flush_writes should not be used, use the proper barrier for whatever this is doing.

Reply:
Why is that not the proper barrier?
We need to make sure the doorbell write is done before we continue with other writes to the LLQ BAR.

Reply:
I think you want to use mmio_wc_start() and mmio_flush_writes() at the end

Reply:
Sorry for the long delay, will fix, thanks.